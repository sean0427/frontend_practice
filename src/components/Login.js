import React from 'react';

const dict = {
    email: 'email',
    password: 'password',
    submit: 'submit',
};

export default class Selector extends React.Component {
    static default = {
        onPress: () => {
            //default do notthing.
        },
    };

    constructor(props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.state = {
            email: '',
            password: '',
        };
    }

    handleSubmit(event) {
        this.props.onPress(this.state.email, this.state.password);
        event.preventDefault();
    }

    handleChange(event) {
        const { name, value } = event.target;

        this.setState({
            [name]: value,
        });
    }

    render() {
        return (
            <form className="login" onSubmit={this.handleSubmit}>
                <lable>
                    <h5>{ dict.email }</h5>
                    <input name="email" type="email" onChange={this.handleChange}/>
                </lable>
                <lable>
                    <h5>{ dict.password }</h5>
                    <input name="password" type="password" onChange={this.handleChange}/>
                </lable>
                <br/>
                <button type="submit">{ dict.submit }</button>
            </form>
        );
    }
}
