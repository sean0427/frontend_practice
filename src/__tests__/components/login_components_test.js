import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Login from '../../components/Login.js';

const LOGIN_FROM_CHILDREND = 3;

describe('test login component', () => {
    it('render defualt', () => {
        const wrapper = shallow(<Login />);

        expect(wrapper.find('.login')).toHaveLength(1);
        expect(wrapper.find('.login').children()).toHaveLength(LOGIN_FROM_CHILDREND);
    });

    it('button click', () => {
        const mockFunction = jest.fn();
        const wrapper = mount(<Login onPress={mockFunction} />);

        wrapper.find('input[name="email"]')
            .simulate('change', { target: { value: 'test@email.test' } });
        wrapper.find('input[name="password"]')
            .simulate('change', { target: { value: 'passworddd' } });
        wrapper.find('form').simulate('submit');

        expect(wrapper.find('button[type="submit"]')).toHaveLength(1);
        expect(mockFunction).toBeCalled();
    });

    it('check change by snapshot', () => {
        const json = renderer.create(<Login />).toJSON();
        expect(json).toMatchSnapshot();
    });
});
