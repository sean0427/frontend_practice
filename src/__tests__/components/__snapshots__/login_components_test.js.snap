// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login component check change by snapshot 1`] = `
<form
  className="login"
  onSubmit={[Function]}
>
  <lable>
    <h5>
      email
    </h5>
    <input
      name="email"
      type="email"
    />
  </lable>
  <lable>
    <h5>
      password
    </h5>
    <input
      name="password"
      type="password"
    />
  </lable>
  <button
    type="submit"
  >
    submit
  </button>
</form>
`;
